src/replica/bulk_load/replica_bulk_loader.cpp

```diff
@@ -428,15 +428,31 @@ error_code replica_bulk_loader::download_sst_files(const std::string &remote_dir
                     remote_dir, local_dir, f_meta.name, fs, f_size);
                 const std::string &file_name =
                     utils::filesystem::path_combine(local_dir, f_meta.name);
-                if (ec == ERR_OK || ec == ERR_PATH_ALREADY_EXIST) {
-                    if (!utils::filesystem::verify_file(file_name, f_meta.md5, f_meta.size)) {
-                        ec = ERR_CORRUPTION;
-                    } else if (ec == ERR_PATH_ALREADY_EXIST) {
+                bool verified = false;
+                if (ec == ERR_PATH_ALREADY_EXIST) {
+                    if (utils::filesystem::verify_file(file_name, f_meta.md5, f_meta.size)) {
                         // local file exist and is verified
                         ec = ERR_OK;
                         f_size = f_meta.size;
+                        verified = true;
+                    } else {
+                        derror_replica(
+                            "file({}) exists, but not verified, try to remove local file "
+                            "and redownload it",
+                            file_name);
+                        if (!utils::filesystem::remove_path(file_name)) {
+                            derror_replica("failed to remove file({})", file_name);
+                            ec = ERR_FILE_OPERATION_FAILED;
+                        } else {
+                            ec = _stub->_block_service_manager.download_file(
+                                remote_dir, local_dir, f_meta.name, fs, f_size);
```

Comment:
remove the `verify_file` below to this line

Reply:
The following cases should verify files:
- ec = ERR_OK
- ec = ERR_PATH_ALREADY_EXIST
- ec = ERR_PATH_ALREADY_EXIST, local file removed, redownload ec = ERR_OK

If `verify_file` move below this line, `ec = ERR_OK` won't be checked

Reply:
But in your code, ec == ERR_OK don't  call the `verify_file` function :
```
                if (ec == ERR_OK && !verified &&
                    !utils::filesystem::verify_file(file_name, f_meta.md5, f_meta.size)) {
                    ec = ERR_CORRUPTION;
                }
```
Only ec == ERR_OK and !verifyed does.

Reply:
Acutually, the possible cases is like below:
1. ec = ERR_OK, verified = false
2. ec = ERR_PATH_ALREADY_EXIST and pass verfication -> ec = ERR_OK, verified = true
3. ec = ERR_PATH_ALREADY_EXIST, local file removed, redownload ec = ERR_OK, verified = false

I add `verified` only to avoid case2 doing verification twice/